ecosystem/attestation-station/using-sdk/README.md

```diff
@@ -0,0 +1,234 @@
+)
+```
+
+This is how you create a [wagmi provider](https://wagmi.sh/core/providers/jsonRpc).
+To use the Optimism production network, you'd use `wagmiChains.optimism` here.
+[See here](https://wagmi.sh/core/chains#supported-chains) to the list of supported chains.
+
+
+```js
+const setup = () => {
+  wagmiCore.createClient({
+    provider,
+    webSocketProvider
+  })
+}   // end of setup
+```
+
+This function sets up [the wagmi client](https://wagmi.sh/core/client).
+Note that this client is a [singleton](https://en.wikipedia.org/wiki/Singleton_pattern), which is why it is not necessary to specify which one we're using later.
+
+This level of connection is sufficient for calling [`view`](https://www.tutorialspoint.com/solidity/solidity_view_functions.htm) functions.
+
+```js
+// \/  Definitions only required for writing attestations  \/
+
+const wagmiCoreMock = import("@wagmi/core/connectors/mock")
+```
+
+When wagmi runs inside a browser, it can use the wallet to prompt the user to sign transactions (using an entity called [`connector`](https://wagmi.sh/examples/connect-wallet)). 
+However, this example runs inside Node, which is a server. 
+There is no user to prompt for signing transactions.
+Therefore, to submit transactions (which is necessary to write attestations), we use a [mock connector](https://wagmi.sh/core/connectors/mock) that has a private key.
+
+If you are going to use `atst` in client-side code to write attestations, use a different connector, probably through [RainbowKit](https://www.rainbowkit.com/docs/installation#configure) of [ConnectKit](https://docs.family.co/connectkit).
+
+```js
+const writeSetup = async () => {
+
+  // Connect the client to a mock connector
+  await wagmiCore.connect({
+    // MockConnector is used for server processes when there is no
+    // user wallet
+    connector: new wagmiCoreMock.MockConnector({
+      options: {
+        chainId: chains[0].ids,
+        signer: new ethers.Wallet(process.env.PRIVATE_KEY, 
+          provider(wagmiChains.optimismGoerli)
+```
+
+A wagmi provider is a function, what gives you [an Ethers `Provider`](https://docs.ethers.org/v5/api/providers/provider/) when you give it the chain you want. 
+
+```js          
+        ),
+      },
+    }),   // end of new wagmiCoreMock.MockConnector
+  })   // end of wagmiCore.connect
+
+}  // end of writeSetup
+
+//  /\  Definitions only required for writing attestations /\
+
+
+const main = async () => {
+
+    setup()
+    await writeSetup()
+```
+
+To write an attestation we need both setups.
+
+```js
+    
+    const readCreatorAddr = "0xc2dfa7205088179a8644b9fdcecd6d9bed854cfe"
+    const aboutAddr = "0x00000000000000000000000000000000000060A7"
+    const key = "animalfarm.school.GPA"
+
+    // Read an attestation
+    const val = await atst.readAttestation(
+        readCreatorAddr,
+        aboutAddr,
+        key,
+        "string")    // data type
```

Comment:
```suggestion
    const val = await atst.readAttestationString(
        readCreatorAddr,
        aboutAddr,
        key)   
```
After watching people struggle with data types this became the recommended approach